`fabric8:build` should check if there is no docker host, then if gofabric8 is on the PATH use `gofabric8 docker-env`

if someone clones a fabric8 enabled project and doesn't install docker or kubernetes, they'll have no `dockerHost` maven property in their `~/.m2/settings.xml` or have `$DOCKER_HOST`

However now we have `mvn fabric8:install` we've java code that can easily check for `gofabric8` on the PATH and download it if required (maybe with a user prompt).

Also if we find `gofabric8` there's a new command soon: `gofabric8 docker-env` which will figure out the docker env vars for minikube or minishift (maybe others later on).

So if there's no DOCKER_HOST the code could run `gofabric8 docker-env` to find the env vars required.

I had a quick look at implementing this - looks like docker-maven-plugin has a private method 
https://github.com/fabric8io/docker-maven-plugin/blob/7e1af5b66871631d4ab3403c282d55f9c6c520d3/src/main/java/io/fabric8/maven/docker/AbstractDockerMojo.java#L298

it would be nice to make that protected so we can override it in the fabric8:build goal to auto-detect docker via gofabric8 instead.

I just tried it. Almost there - it does detect DOCKER_HOST now:
```
[INFO] F8> DOCKER_HOST from gofabric8 : tcp://192.168.64.81:2376
2016-09-26 09:10:26 INFO  RetryExec:96 - I/O exception (org.apache.http.conn.UnsupportedSchemeException) caught when processing request to {}->tcp://192.168.64.81:2376: tcp protocol is not supported
2016-09-26 09:10:26 INFO  RetryExec:112 - Retrying request to {}->tcp://192.168.64.81:2376
2016-09-26 09:10:26 INFO  RetryExec:96 - I/O exception (org.apache.http.conn.UnsupportedSchemeException) caught when processing request to {}->tcp://192.168.64.81:2376: tcp protocol is not supported
2016-09-26 09:10:26 INFO  RetryExec:112 - Retrying request to {}->tcp://192.168.64.81:2376
2016-09-26 09:10:26 INFO  RetryExec:96 - I/O exception (org.apache.http.conn.UnsupportedSchemeException) caught when processing request to {}->tcp://192.168.64.81:2376: tcp protocol is not supported
2016-09-26 09:10:26 INFO  RetryExec:112 - Retrying request to {}->tcp://192.168.64.81:2376
[ERROR] F8> Cannot create docker access object
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 10.082 s
[INFO] Finished at: 2016-09-26T09:10:26+01:00
[INFO] Final Memory: 39M/477M
[INFO] ------------------------------------------------------------------------
[WARNING] The requested profile "quickstarts" could not be activated because it does not exist.
[ERROR] Failed to execute goal io.fabric8:fabric8-maven-plugin:3.1-SNAPSHOT:build (fmp) on project demo: Cannot create docker access object: Cannot extract API version from server tcp://192.168.64.81:2376: tcp protocol is not supported -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```
Not sure what the issue is. If I ran `eval $(gofabric8 docker-env)` it was all fine. I wonder if the java code needs to interpret `tcp://` as `http://` (have some vague memory the java code prefers http:// URLs

--- a/core/src/main/java/io/fabric8/maven/core/util/Gofabric8Util.java
+++ b/core/src/main/java/io/fabric8/maven/core/util/Gofabric8Util.java
@@ -22,6 +22,7 @@ import java.util.*;
 
 import io.fabric8.maven.docker.access.DockerConnectionDetector;
 import io.fabric8.maven.docker.access.util.EnvCommand;
+import io.fabric8.maven.docker.util.EnvUtil;
 import io.fabric8.maven.docker.util.Logger;
 
 /**
@@ -62,9 +63,11 @@ public class Gofabric8Util {
                 envMap = command.getEnvironment();
             }
             String dockerHost = envMap.get("DOCKER_HOST");
-            if (dockerHost != null) {
-                log.info("DOCKER_HOST from gofabric8 : %s",dockerHost);
+            if (dockerHost == null) {
+                return null;
             }
+            dockerHost = EnvUtil.convertDockerHostToUrl(dockerHost);
+            log.info("DOCKER_HOST from gofabric8 : %s", dockerHost);
             return dockerHost;
         }
 

--- a/parent/pom.xml
+++ b/parent/pom.xml
@@ -61,7 +61,7 @@
     <fabric8.maven.plugin.version>3.1.51</fabric8.maven.plugin.version>
     <version.maven>3.3.1</version.maven>
     <version.fabric8>2.2.161</version.fabric8>
-    <version.docker-maven-plugin>0.16.3</version.docker-maven-plugin>
+    <version.docker-maven-plugin>0.16.4</version.docker-maven-plugin>
   </properties>
 
   <repositories>

